docs/graphcache/gotchas.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Gotchas
+order: 5
+---
+
+# Gotchas
+
+When using this normalised cache there are a few out of the box features
+we need to be aware of, being aware of these features will make it easier
+to define best practices and to avoid pitfalls.
+
+Most of the measures we'll cover are brought to life to ensure we can build out
+to a fully offline normalised cache.
+
+## Commutativity
+
+By default the cache ensures idempotent and commutative results, the last part of this
+is the most relevant one to be aware of.
+
+Commutative guarantees mean that we'll keep the order of operations consistent, let's look
+at an example to clear this up.
+We are in our application and we queue up two queries, an `authorsQuery` and a `todosQuery`.
+
+When we dispatch `todosQuery` first and `authorsQuery` later but the `authorsQuery` returns a result
+before the `todosQuery` we'll put this result in a layer (this means it isn't actually in the real data)
+until the `todosQuery` arrives. When the `todosQuery` arrives we'll commit that first and then the `authorsQuery`
+to ensure our results are accurate. 
+
+## Optimistic results & refetches
+
+When we have mutations in-flight that have optimistic entries we defer refetches due to `cache-and-network` and
```

Comment:
```suggestion
When mutations with optimistic entries are in-flight, refetches are deferred due to `cache-and-network` and
```

Minor tweak to reduce the amount uses of "we"

Reply:
We can probably avoid this by phrasing this with an explanation first?

> Optimistic updates can temporarily update the data after incoming mutations, which will trigger on screen queries to update. However, if we also use `cache-and-network` at the same time, some queries can refetch and overwrite our optimistic data, causing an unintended state where the intended optimistic update is destroyed.
> Such an unintended refetch can also happen if after an optimistic update a query is refetched when it’s not or only partially available in the cache.
> 
> To prevent this, Graphcache will temporarily pause refetches that may overwrite optimistic updates. Once all mutations with optimistic updates complete however, all results will be applied at once, and refetched that may update the mutation data will be rerun.